source/vibe/web/rest.d

```diff
@@ -433,18 +433,22 @@ class RestInterfaceClient(I) : I
 			auto query = appender!string();
 
 			foreach (idx, p; params) {
-				static if (!(idx % 2)) {
-					// Parameter name
-					static assert(is(typeof(p) == string), "Internal error ("~__FUNCTION__~"): Parameter name is not string.");
-					query.put('&');
-					filterURLEncode(query, p);
-				} else {
-					// Parameter value
-					query.put('=');
-					static if (is(typeof(p) == Json))
-						filterURLEncode(query, p.toString());
-					else // Note: CTFE triggers compiler bug here (think we are returning Json, not string).
-						filterURLEncode(query, toRestString(serializeToJson(p)));
+				// gets value corresponding to the current parameter
+				auto value = params[(idx / 2) * 2 + 1];
+				// generates condition for inclusion of this parameter
+				enum condition = isNullable!(typeof(value)) ? q{ !value.isNull } : q{ true };
+				if (mixin(condition))
+				{
+					static if (!(idx % 2)) {
+						// Parameter name
+						static assert(is(typeof(p) == string), "Internal error ("~__FUNCTION__~"): Parameter name is not string.");
+						query.put('&');
+						filterURLEncode(query, p);
+					} else {
+						// Parameter value
+						query.put('=');
+						filterURLEncode(query, toRestString(p));
+					}
```

Comment:
[0, 1, 2, 3, 4, 5] shouldn't result in [1, 3, 3, 5, 5, 5], but in [1, 1, 3, 3, 5, 5], therefore indexing every odd element, which would be the value of the parameter that's currently being processed. 

This is needed so that the code can decide whether or not it should emit the parameter if it's an nullable. And this decision must be done for both the parameter name and the parameter value.

Although I do agree that it's a bit unclear. I'm thinking of rewriting it so that the foreach processes only even (or odd, doesn't really matter) elements and then handle both the name and value at the same time.